src/storage-types/src/sources/kafka.rs

```diff
@@ -87,27 +88,38 @@ pub static KAFKA_PROGRESS_DESC: Lazy<RelationDesc> = Lazy::new(|| {
         .with_column("offset", ScalarType::UInt64.nullable(true))
 });
 
-impl<C: ConnectionAccess> KafkaSourceConnection<C> {
+impl KafkaSourceConnection {
     /// Returns the client ID to register with librdkafka with.
     ///
     /// The caller is responsible for providing the source ID as it is not known
     /// to `KafkaSourceConnection`.
-    pub fn client_id(&self, connection_context: &ConnectionContext, source_id: GlobalId) -> String {
-        format!(
+    pub fn client_id(
+        &self,
+        configs: &ConfigSet,
+        connection_context: &ConnectionContext,
+        source_id: GlobalId,
+    ) -> String {
+        let mut client_id = format!(
             "materialize-{}-{}-{}",
             connection_context.environment_id, self.connection_id, source_id,
-        )
+        );
+        self.connection.enrich_client_id(configs, &mut client_id);
+        client_id
     }
+}
 
+impl<C: ConnectionAccess> KafkaSourceConnection<C> {
     /// Returns the ID for the consumer group the configured source will use.
     ///
     /// The caller is responsible for providing the source ID as it is not known
     /// to `KafkaSourceConnection`.
     pub fn group_id(&self, connection_context: &ConnectionContext, source_id: GlobalId) -> String {
         format!(
-            "{}{}",
+            "{}materialize-{}-{}-{}",
             self.group_id_prefix.as_deref().unwrap_or(""),
-            self.client_id(connection_context, source_id)
+            connection_context.environment_id,
+            self.connection_id,
+            source_id,
```

Comment:
does it make sense to impl a common method (perhaps on `KafkaConnection`) that generates this `materialize-{}-{}-{}` string? We now do the same formatting in 5 places across `src/storage-types/src` - might be worth centralizing

Reply:
I went back and forth on it. Part of what's weird is that `connection_id` is only available on the source/sink types, not the underlying connection, and so making the helper method part of the connection awkwardly requires passing in the connection's own ID. 🤷🏽‍♂️ I'll hammer on it a bit.

Reply:
Extracted a `KafkaConnection::id_base` helper method!